src/storage/src/tests/byte_array_test.cairo

```diff
@@ -0,0 +1,202 @@
+use starknet::{ClassHash, ContractAddress, SyscallResult, SyscallResultTrait};
+
+#[starknet::interface]
+trait IByteArrayStorage<T> {
+    fn get_b1(self: @T) -> ByteArray;
+    fn set_b1(ref self: T, value: ByteArray);
+    fn get_b2(self: @T) -> ByteArray;
+    fn set_b2(ref self: T, value: ByteArray);
+    fn get_map_value(self: @T, key: felt252) -> ByteArray;
+    fn set_map_value(ref self: T, key: felt252, value: ByteArray);
+    fn get_map_key(self: @T, key: ByteArray) -> felt252;
+    fn set_map_key(ref self: T, key: ByteArray, value: felt252);
+    fn health(self: @T) -> felt252;
+}
+
+#[starknet::contract]
+mod byte_array_storage {
+    use alexandria_storage::byte_array::{LegacyHashByteArray, StoreByteArray};
+
+    #[storage]
+    struct Storage {
+        b1: ByteArray,
+        b2: ByteArray,
+        map_value: LegacyMap<felt252, ByteArray>,
+        map_key: LegacyMap<ByteArray, felt252>,
+    }
+
+    #[external(v0)]
+    impl ImplByteArrayStorage of super::IByteArrayStorage<ContractState> {
+        fn get_b1(self: @ContractState) -> ByteArray {
+            self.b1.read()
+        }
+
+        fn set_b1(ref self: ContractState, value: ByteArray) {
+            self.b1.write(value);
```

Comment:
If the signature of the `write` function changes we want to change this one also no?
```suggestion
            self.b1.write(value)
```

Reply:
Fixed in 827b8c889a291fbf8d84daa08d4b3b3643b8b418. That's a good point, but we will anyway need to update the return type, correct? Or sorry if I don't get the point well.

Reply:
exactly